### DISCUSSION TITLE
Local Plugin Support

### DISCUSSION THREAD

**Discussion by ghost** [Flow-Launcher/Flow.Launcher#2175] [Ideas]
Hey Folks,

There could be a compelling use case to use this as an internal tool at some organizations. If I wanted to create a local plugin and have that bundled with my FlowLauncher is that supported? If I wanted to do this, I imagine I would have to replicate some of the GitHub Action steps locally and bundle my plugins. 

I see a strong usecase for local plugins because there could be some internal tools an organization wants to hoist but not make public to the larger world.

**Comment by jjw24** [comment]
>If I wanted to create a local plugin and have that bundled with my FlowLauncher is that supported?

When you say bundled, do you mean like for example the Explorer plugin that is packaged up inside the installer?

**Reply by jjw24** [reply]
If this is the case, adding the local plugin project to flow's main project (like the other default plugins) will allow it to be packaged up inside the installer. It will work without much changes if hosted on GitHub (you will need to set it up with an AppVeyor account), otherwise yes you will need to replicate the AppVeyor pipeline file (this is the platform we build and deploy flow).

**Reply by ghost** [reply]
Yea, we have gone with this approach and it seems to work o-k. The plan is to use GitHub eventually and rolling out the local server and have our plugins hosted there. For now, this suffices.

**Comment by jjw24** [comment]
There is a new feature we want to add which is a better solution for your organisation's use case- we want to let the plugins manifest URL be configurable from the settings UI, so organisation like yours can then just configure flow's Plugin Store to load your privately hosted plugins instead.

https://github.com/Flow-Launcher/Flow.Launcher/issues/2178

**Reply by ghost** [reply]
That would be awesome! That would make things a lot easier. Pardon my ignorance, but assuming one did not have access to github actions ( yet ), what would you recommend for local plugin development?

**Comment by VictoriousRaptor** [comment]
`pm install` supports URLs so an alternative way is to store your local plugin in an internal URL and let your users install it. Though users won't get update:)

**Comment by ghost** [comment]
Thank you all for your suggestions, i'll try and do a PoC early next week and come back with what i've learned. Hopefully can use that learning lesson to expand the current ( great ) guide.

**Comment by ghost** [comment]
Hi @jjw24, thanks for your help! Do you know where I can track the External Plugin Server? We have some cool PoC's working locally that we would like to hoist on a plugin server.

**Reply by jjw24** [reply]
Hi there, so all actual plugins are hosted on GitHub. Flow relies on a plugin manifest file to tell it where to download the selected plugin. This manifest lives at https://github.com/Flow-Launcher/Flow.Launcher.PluginsManifest/blob/plugin_api_v2/plugins.json.

Hope this helps.

**Reply by ghost** [reply]
Helps a lot, thank you!
